### PR TITLE
codebase example

### PR DESCRIPTION
**Pull request by gloverd2**
codebase example & change to namespace to allow use of expl_categorical in notebook

### CODE REVIEW

**Review by clementstross — CHANGES_REQUESTED** [review]
https://ourcodingclub.github.io/tutorials/writing-r-package/

can we move this to a notebook folder (or somewhere else sensible and add it to the .Rbuildignore).

* Can we have a better name than code
* Needs to have the suffix .ipynb for me to open it
* I haven't built the package so can't install it. Can we install it from the repo?
* Can we get the data in the repo? This is pretty normal to include example data. Loading it from file means a new person can't recreate the notebook
* I'd like some markdown text saying what each stage is trying to do. Have you used notebooks before or would you like me to show you how to do it?
* for ADMR::expl_categorical i'd add text saying this output could very easily be plotted
* Some of the lower cells don't have output saved.

Would a call help go over some of these points?

**Comment by gloverd2** [comment]
1.Can we have a better name than code - **example_script?**

2. Needs to have the suffix .ipynb for me to open it - **that is does, i very much didnt notice my type there**

3. I haven't built the package so can't install it. Can we install it from the repo? **Should be able to build from the Master Branch** 

4. Can we get the data in the repo? This is pretty normal to include example data. Loading it from file means a new person can't recreate the notebook. **I was just reading up on that. doesnt seem too difficult** 

5. I'd like some markdown text saying what each stage is trying to do. Have you used notebooks before or would you like me to show you how to do it? **to do list** 

6. for ADMR::expl_categorical i'd add text saying this output could very easily be plotted **can be included - todo list**

7. Some of the lower cells don't have output saved. - ** I didn't actually run them beforehand oops **

**Review by clementstross — APPROVED** [review]
I've made changes to this branch so my review means less. I'm happy that Dave and I have come to an agreement so I'm willing to approve and merge. @gloverd2 I'll leave it to you to decide if someone else should look at it or if we should merge

Although lets remove 
.ipynb_checkpoints/example_script-checkpoint.ipynb

File: codebase_example_script.Rmd [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+```{r Package Functions, echo=FALSE}
+ls("package:ADMR")
+```
+### Import 'Predicting Household Sale Prices' dataset
+```{r Import data, echo=TRUE}
+data(hh_train)
+data(hh_test) 
+```
+
+```{r process data, include=FALSE}
```
[the remark this answers is not in the document]
Comment (gloverd2) [reply codebase_example_script.Rmd:49]: that was more for me to see what was going on at the time. 
I dont particularly think we need to show the names in the output.
but i dont mind if we do either.

File: codebase_example_script.Rmd [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+### encode_freq
+Apply frequency encoding with the option to group the rarest levels
+```{r include=FALSE}
+hh_train$Neighborhood[5:10] <- NA
+```
+```{r encode_freq, echo=TRUE}
+table(hh_train$Neighborhood, exclude = NULL)
+nhood_freq <- ADMR::encode_freq(data = hh_train$Neighborhood,
```
[the remark this answers is not in the document]
Comment (gloverd2) [reply codebase_example_script.Rmd:94]: I think you have done this already?

**Pull request merged**